### PR TITLE
Add openfoam taskflow

### PR DESCRIPTION
**Pull request by jourdain**

### CODE REVIEW

**Comment by jourdain** [comment]
Might be good enough to be merge before a global cleanup between taskflows...

File: server/taskflows/hpccloud/taskflow/openfoam/DockerOpenFOAM_run.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+cd $1
+
+# Env setup
+source /opt/OpenFOAM/OpenFOAM-v1612+/etc/bashrc
```
Comment (cjh1) [inline server/taskflows/hpccloud/taskflow/openfoam/DockerOpenFOAM_run.sh]: The /opt/OpenFOAM.... path should probably be cluster configuration parameter for the workflow?

File: server/taskflows/hpccloud/taskflow/openfoam/motorbike.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright 2016 Kitware Inc.
```
Comment (cjh1) [inline server/taskflows/hpccloud/taskflow/openfoam/motorbike.py]: 2017

File: server/taskflows/hpccloud/taskflow/openfoam/motorbike.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                "id": <the id of the folder containing input files>
+            },
+        },
+        "dataDir": <passed to --data-dir,
+        "cluster": {
+            "_id": <this id of the cluster to run on>
+        },
+        "sessionKey": <a unqiue key to use to store the connection information>,
+        "output": {
+          "folder": {"id": <the id of a folder to upload any output> }
```
Comment (cjh1) [inline server/taskflows/hpccloud/taskflow/openfoam/motorbike.py]: Not sure this doc string was updated?

File: server/taskflows/hpccloud/taskflow/openfoam/tutorial.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+            'folderId': output_folder_id,
+            'path': '.'
+        }]
+    else:
+        task.taskflow.logger.info('No output.folder.id')
+
+    upload_job_output_to_folder(cluster, job, log_write_url=None, job_dir=None,
+                                girder_token=task.taskflow.girder_token)
+
+    task.taskflow.logger.info('Upload complete.')
```
Comment (cjh1) [inline server/taskflows/hpccloud/taskflow/openfoam/tutorial.py]: This seems almost identical to motobike.py?

File: server/taskflows/hpccloud/taskflow/openfoam/DockerOpenFOAM_run.sh (hunk 1 above)
Comment (jourdain) [inline server/taskflows/hpccloud/taskflow/openfoam/DockerOpenFOAM_run.sh]: Not really that's the docker path. For a real OpenFOAM deployment such that we can use MPI, more configuration will be needed. But not here.

File: server/taskflows/hpccloud/taskflow/openfoam/motorbike.py (hunk 3 above)
Comment (jourdain) [inline server/taskflows/hpccloud/taskflow/openfoam/motorbike.py]: Most likely not. Will go through and clean that.

File: server/taskflows/hpccloud/taskflow/openfoam/tutorial.py (hunk 4 above)
Comment (jourdain) [inline server/taskflows/hpccloud/taskflow/openfoam/tutorial.py]: yes, that's expected.

File: server/taskflows/hpccloud/taskflow/openfoam/DockerOpenFOAM_run.sh (hunk 1 above)
Comment (cjh1) [inline server/taskflows/hpccloud/taskflow/openfoam/DockerOpenFOAM_run.sh]: OK, make sense if that is in the docker container

File: server/taskflows/hpccloud/taskflow/openfoam/tutorial.py (hunk 4 above)
Comment (cjh1) [inline server/taskflows/hpccloud/taskflow/openfoam/tutorial.py]: So shouldn't we use parameters for the difference and reuse the code?

Comment (jourdain) [inline server/taskflows/hpccloud/taskflow/openfoam/tutorial.py]: Well we can trash the motorbike.py. That was my first test bed when simput was not yet involved.

Comment (cjh1) [inline server/taskflows/hpccloud/taskflow/openfoam/tutorial.py]: Ok, sounds good

**Comment by jourdain** [comment]
This should be fine too now. Can you review it? @cjh1

**Pull request merged**